### DISCUSSION TITLE
Chapter titles for mp3 files generated with AAC

### DISCUSSION THREAD

**Discussion by mettavi** [audiamus/AaxAudioConverter#143] [Q&A] [answered]
Hi, firstly I would like to thank you for all the work you have put into this software.

I have a problem which I hope you can help me with. I have searched the forum but have been unable to find anything that looks relevant.

My main goal is to get descriptive chapter titles from the AAX file. I downloaded it using Book Lib Connect. It worked perfectly when generating a single file. 

However, I also wanted to have a version in multiple mp3 files for compatibility. I chose the third output option, "Multiple files, by chapter, split into shorter tracks." This generated multiple folders with mp3 files and a .m3u playlist file.  The folders are named according to the descriptive chapter titles, however, the individual mp3 files are not. The title metadata is also the same as the filenames. Thus, when I play the playlist in VLC (on a Mac), all I see are the generic numbered chapter files. Here is a screenshot:

<img width="877" alt="mp3 playlist" src="https://user-images.githubusercontent.com/1978429/187600414-469b6fd8-81e4-480f-b5f5-8ecf06ff4ae1.png">

I am not sure if this is a problem with AAC, VLC, or perhaps I have not configured the software correctly. 

Is there a way to export the files using AAC so that the m3u playlist displays the named chapter titles in a media player like VLC?

Regards,
scop22.

**Comment by audiamus** [comment]
Descriptive chapter names depend on the extra JSON files. If names are present there, you can add them to both title tags and file names.
These are the settings I use for my books:

- File naming pattern: track - book - author
- Title tag naming pattern: track - book
- Track numbering pattern: track(chapter)
- Add total tracks: yes

Play around with the settings and read up on the description for each item in the manual.

**Reply by mettavi** [reply]
Hi Audiamus, many thanks for your reply and once again for developing this excellent software. 👍 

I had a look at the options you mentioned, but I still can't work out how to do what I want. Let me try to explain more with reference to this screenshot:

![Split chapter names](https://user-images.githubusercontent.com/1978429/188258996-4917c782-0f1c-48c0-b299-c062b113dfbc.jpg)

You can see here that the folders have the descriptive chapter names, but the individual files don't. I would like the descriptive chapter names to show up in my media player (currently using VLC on Mac) using the m3u playlist so that it is easier to navigate to different sections (this is a long audiobook, about 60 hours I think). So, for example, I would like the individual file in the screenshot to be titled "165 - Chapter 2 Walking" and so on.

Using the file naming/title tag naming pattern options, what would I select to achieve this? AFAICT, "track-book-author" will just produce the filenames as shown in the screenshot, and not the folder names.

Regards, scop22.

**Comment by audiamus** [comment, marked as answer]
Again, I suggest to try my settings from the previous post. The setting for the "track" token will affect its usage for the file name and title tag.

So, if the "track" setting is "track(chapter)" and the "title tag" is "track - book", for track # 34 and chapter title "Chapter 7: The 7th Chapter" the result in the title tag will be "34(Chapter 7: The 7th Chapter) - My Great Book". Similar with the file naming pattern. 

(There are few restrictions with file names as not all characters are valid for them, but the app will take care of that.)

**Reply by mettavi** [reply]
Ah, I see. I didn't realise that the track setting was a variable that gets plugged into the other settings. When I tried it as you suggested, it worked. 

Many thanks for the help.
